### ISSUE TITLE
Installation fails due to orjson dependency 

### ISSUE THREAD

**Issue by Mtze** [readthedocs/readthedocs.org#7687]
## Details

* Read the Docs project URL:
* Build URL (if applicable):
* Read the Docs username (if applicable):

I followed the install instructions here (https://docs.readthedocs.io/en/stable/development/install.html), but the requirements step fails with
```
Collecting orjson==2.0.7
  Downloading orjson-2.0.7.tar.gz (480 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp4ejl15q6
         cwd: /tmp/pip-install-v9upuffk/orjson
    Complete output (6 lines):

    Cargo, the Rust package manager, is not installed or is not on PATH.
    This package requires Rust and Cargo to compile extensions. Install it through
    the system's package manager or via https://rustup.rs/

    Checking for Rust toolchain....
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp4ejl15q6 Check the logs for full command output.
```

## Expected Result

Requirements get installed

## Actual Result

The desired version of orjson (2.0.7) cannot be installed.

**Comment by Mtze** [comment]
Maybe remove the explicit version number here https://github.com/readthedocs/readthedocs.org/blob/5234f41de5db55029eba7bf3d5f400c4aa7bb891/requirements/pip.txt#L64 except it is really necessary.

**Comment by humitos** [comment]
Hi! We currently support Python 3.6 and it works with the current orjson defined in the requirements. You may want to try that.

We are tracking this issue in https://github.com/readthedocs/readthedocs.org/issues/7313 --please subscribe there for news on this.

**Comment by Split-way** [comment]
Had the same problem on Ubuntu 23.04, Python 3.11 and orjson 3.6.7. The issue turned out to be in an python3.11-incompatible orjson version. Fixed by updating orjson to Python 3.11-compatible version - 3.7.12.
